src/scripts/main.js

```diff
@@ -1,5 +1,33 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const spiderW = spider.clientWidth;
+  const spiderH = spider.clientHeight;
+  const wall = document.querySelector('.wall');
+
+  if (!e.target.matches('.wall')) {
+    return;
+  }
+
+  const spiderX = e.clientX - wall.getBoundingClientRect().x
+    + window.pageXOffset - spiderW / 2 - wall.clientLeft;
+
+  const spiderY = e.clientY - wall.getBoundingClientRect().y
+    + window.pageYOffset - spiderH / 2 - wall.clientTop;
+
+  function checkInside(coord, spiderSize, max) {
+    let result = coord;
+
+    if (coord + spiderSize > max) {
+      result = max - spiderSize;
+    } else if (coord < 0) {
+      result = 0;
+    }
+
+    return result;
+  }
```

Comment:
move outside the eventListener, now you create a new function every time you click on document 0_0

Reply:
checkInside? it's not obvious what this function is for